internal/workload/manager.go

```diff
@@ -234,6 +234,82 @@ func (w *WorkloadManager) RegisterObserver(observer Observer) {
 	w.workloads.RegisterObserver(observer)
 }
 
+func (w *WorkloadManager) Deregister() error {
+	w.managementLock.Lock()
```

Comment:
if we obtain the lock here, and from some reason in another thread, `workload.Update()` is waiting on the lock to be released, the call for `workload.Update()` will fail (either on attempt to read manifest folder, or at other step of its execution).
Eventually the error message will be propagated as warning to the logs.
But it can be avoided.
Also, we need to terminate `ensureWorkloadsFromManifestsAreRunning()` which will fail to read manifest folder

Reply:
why? 
If another thread `workload.Update()` will wait till the lock would be releases.
Then - the section of https://github.com/jakub-dzon/k4e-device-worker/blob/537d75ce04f05eddbbbd6859bfad0877b9170688/internal/workload/manager.go#L81 would not be run (there isn't any workloads) and also deployedWorkloadByName would be empty so the section  https://github.com/jakub-dzon/k4e-device-worker/blob/537d75ce04f05eddbbbd6859bfad0877b9170688/internal/workload/manager.go#L121
would not be run also....

Reply:
Do you refer to the finalizers that assures the disconnect is being called only if YggdrasilWorkloadFinalizer was removed ?
In that case - there is a point. That would be the responsibility of the operator not to send any deployments to the device if it has not YggdrasilWorkloadFinalizer finalizer.

Reply:
I added a boolean for that case. 
Please take a look